Improve user install experience

rashmihunt opened the pull request:
This is a first step in improving user install experience in installing this connector. The improvement is, during installation, if pre-packaged oracle install client pre-req tarball is not found, then code checks if user has installed oracle prereqs offline and attempts to install oracledb to complete the installation, else it indicates the user to install the pre-reqs as per oracledb isntallation guide based on their OS.

@raymondfeng PTAL

raymondfeng commented on lib/download.js:
```diff
@@ -71,49 +72,75 @@ module.exports = function (url, version, dest, cb) {
+    //traverse to the apiconnct project level and install oracledb. This will also work in loopback-connector-oracle dev env
+    //where loopback-oracle-installer is installed as dependent module to loopback-connector-oracle. In this case, oracledb
+    //is installed as dependent module of  loopback-connector-oracle.
+    var dirName =  path.join(__dirname, '..', '..', '..');
```
We probably should use `require.resolve('loopback-connector-oracle/package.json')` to locate the oracle connector module.

rashmihunt commented:
List of changes
1) Moved oracle related config to installer
2) In apiconnect project env, loopback-oracle-installer is a pier of loopback-connector-oracle. Oracledb will be installed as a child of  loopback-connector-oracle.
3) In loopback-connector-oracle dev environment, loopback-oracle-installer is a child of loopback-connector-oracle. Oracledb will be installed as a child of  loopback-connector-oracle. This is consistent to point 2.
4) In loopback-oracle-installer dev environment, loopback-connector-oracle doesn't exist. Oracledb gets installed as a child of loopback-oracle-installer. 
5) Cleaned up the code
6) Removed usage of __dirName

@raymondfeng PTAL

rashmihunt commented:
@slnode test please

raymondfeng commented on package.json:
```diff
@@ -27,7 +28,10 @@
+    "excludeAbi": "11",
+    "oracleUrl": "http://7e9918db41dd01dbf98e-ec15952f71452bc0809d79c86f5751b6.r22.cf1.rackcdn.com",
+    "oracleVersion": "2.0.0",
+    "driverModule": "oracledb@^1.11.0"
```
Now I don't think it is ideal to have the configuration in the installer module for the following reasons:

1. We should be able to use the same version of `loopback-oracle-installer` for different versions of `loopback-connector-oracle`.

2. With the proposed change, we'll have create a new release of the installer every time we need to update the connector.

rashmihunt replied:
So, reverting back to what was before..  
 "config": {
    "excludeAbi": "11"
  } remains in oracle installer as before and rest of the config will be reverted back to oracle connector?

rashmihunt commented:
@raymondfeng PTAL

rashmihunt commented:
@slnode test please

rashmihunt commented:
@slnode test please
